Library/Homebrew/missing_formula.rb

```diff
@@ -187,6 +190,16 @@ def deleted_reason(name, silent: false)
         end
       end
 
+      def cask_reason(name, silent: false)
+        return if silent
+
+        cask = Cask::CaskLoader.load(name)
+        reason = "Found the following cask named \"#{name}\" instead:\n"
+        reason << Cask::Cmd::Info.get_info(cask)
```

Comment:
Testing this out: I'm not sure what I think about outputting the `brew cask info` output if you `brew install firefox` and it fails; I'm interested what other @Homebrew/maintainers think. It might be enough in that case to just output the `Found the following cask...` and skip the `brew cask info` bit (but still output that at `brew info` time).

Reply:
Agreed. If I tried an `install` command and got an `info` output, I’d be confused and start wondering what I typed wrong. Yes, the reason is there, but it could get lost in the rest of the (larger, familiar, and ignorable) `info` text.

Since this equates to a failed command, it should output only the minimum amount of relevant information to “fix”.

Reply:
I pushed a solution that has MissingFormula.cask_reason being called only during the `brew info` command, let me know what you think.

Reply:
@zachauten I think it would still be good to print out that a Cask was found but just not the `brew info` output. You could add an argument to `def reason` which is passed to `cask_reason` to conditionally output the `brew cask info` output.